### PR TITLE
win condition: diagonal_win_left_bottom

### PR DESCRIPTION
**Pull request by OKohler1991**
changed player input for better debugging

changed the whole def diagonal_win_left_bottom 
    simplified the logic and bug fixing
    e.g.: if the player want to set a token on "41", it was set on "31"

### CODE REVIEW

File: 211018_4win_main.py [hunk 1]
```diff
@@ -49,28 +49,21 @@ def digonal_win_left_top(player):
...
-                    print("+")
-                    count += 1
+    for column in range(7):
+        for row in range(5, 0 - 1, -1):             # begin to look after a player token on position "50"
+            if matrix[row][column] == player:
+                for token in range(4):             # begin to check, if the token begins a connect4
+                    if row < 3 or column > 3:       # on position "33": last chance to begin a connect4 from left bottom to right up 
+                        return True
+                    if matrix[row - token][column + token] == player:   # check if their are other player tokens on the up right diagonal
+                        count += 1
```
Comment (Ionenstrahl) [inline 211018_4win_main.py:59]: Nice Olli,
I like the appoach.
It took me some mind bending and Im confinced that there is even shorter version out there, but for now: I agree with your changes.
They are way less hacky than the n-variable I convinced Leo to use

File: 211018_4win_main.py [hunk 2]
```diff
@@ -49,28 +49,21 @@ def digonal_win_left_top(player):
...
-                if matrix[row][column + n] == player:
-                    print("+")
-                    count += 1
+    for column in range(7):
+        for row in range(5, 0 - 1, -1):             # begin to look after a player token on position "50"
+            if matrix[row][column] == player:
+                for token in range(4):             # begin to check, if the token begins a connect4
+                    if row < 3 or column > 3:       # on position "33": last chance to begin a connect4 from left bottom to right up 
+                        return True
+                    if matrix[row - token][column + token] == player:   # check if their are other player tokens on the up right diagonal
```
Comment (Ionenstrahl) [inline 211018_4win_main.py:58]: typo in the comment:
should be "there"

File: 211018_4win_main.py [hunk 3]
```diff
@@ -93,9 +86,9 @@ def row_win(player):
     print("start")
     for row in range(6):
         count = 0
-        for index in range(7):
-            print(row, index)
-            if matrix[row][index] == player:
+        for column in range(7):
+            print(row, column)
```
Comment (Ionenstrahl) [inline 211018_4win_main.py:90]: Leo's gonna be mad, that you are removing his beloved index :D
But I agree with the changes. They make it more readable.

File: 211018_4win_main.py [hunk 4]
```diff
@@ -117,17 +110,18 @@ def row_win(player):
 y = 5
 while play:
     if turn % 2 != 0:
-        inputP1 = int(input("_"))
+        inputP1 = int(input("Player 1 "))
```
Comment (Ionenstrahl) [inline 211018_4win_main.py:113]: It's more descriptive, but in the end needs to be adjusted again.
Codewars test do not like inputs.

**Review by Ionenstrahl — APPROVED** [review]
I agree with Ollis changes
@opencurlybrackets @Gaskarum

File: 211018_4win_main.py [hunk 5]
```diff
@@ -49,28 +49,21 @@ def digonal_win_left_top(player):
...
-        n = -2
-        for row in range(6):
-            if column + n == 0:
-                count = 0
-            try:
-                if matrix[row][column + n] == player:
-                    print("+")
-                    count += 1
+    for column in range(7):
+        for row in range(5, 0 - 1, -1):             # begin to look after a player token on position "50"
```
Comment (Ionenstrahl) [inline 211018_4win_main.py:53]: I dont see why you use "range(5, 0 - 1, -1)"
"range(6)" yields the same results and is more readable

File: 211018_4win_main.py (hunk 4 above)
Comment (OKohler1991) [reply 211018_4win_main.py:113]: yeah, I changed it for easier debugging

File: 211018_4win_main.py (hunk 5 above)
Comment (OKohler1991) [reply 211018_4win_main.py:53]: Yeah, it does.
I didn't thought about it, as I wrote this part of the code

File: 211018_4win_main.py (hunk 3 above)
Comment (opencurlybrackets) [reply 211018_4win_main.py:90]: ahhhhhh index!!!!!!!!!!

File: 211018_4win_main.py (hunk 2 above)
Comment (opencurlybrackets) [reply 211018_4win_main.py:58]: try except the best

**Pull request merged**